viewing_party/party.py

```diff
@@ -1,23 +1,248 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title == None:
+        return None
+    elif genre == None:
+        return None
+    elif rating == None:
+        return None
+    else:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    movie_index = -1
+
+    # find index of watched movie in watchlist list
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == title:
+            movie_index = i
+
+    # if movie is found to be in watchlist
+    if movie_index >= 0:
+        # remove element at that index from watchlist list and add to variable
+        watched_movie = user_data["watchlist"].pop(movie_index)
+
+        # add variable to watched list
+        user_data["watched"].append(watched_movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    """
+    Returns user's average movie rating.
+    """
+    sum = 0
+    average = 0
+    watched_list_length = len(user_data["watched"])
+
+    if watched_list_length > 0:
+        # total up ratings
+        for i in range(watched_list_length):
+            sum += user_data["watched"][i]["rating"]
+        # divide total by number of movies to get average
+        average = sum / watched_list_length
 
+    return average
+
+def get_most_watched_genre(user_data):
+    """
+    Returns user's most watched genre.
+    """
+    if len(user_data["watched"]) > 0:
+        # make new list for just genres, with repetitions
+        watched_genre_list = []
+        for i in range(len(user_data["watched"])):
+            watched_genre_list.append(user_data["watched"][i]["genre"])
+        
+        most_watched_count = 0
+        most_watched_genre = ''
+
+        # iterate through genre list, counting how many times each genre appears
+        for genre in watched_genre_list:
+            genre_count = watched_genre_list.count(genre)
+            # update most_watched_genre and genre count with whatever comes back highest
+            if genre_count > most_watched_count:
+                most_watched_count = genre_count
+                most_watched_genre = genre

```

Comment:
Nice approach to flatten the list so you can use count on it! Another way to get our count would be to use the `max` function on the list of genres:
```suggestion
        most_watched_genre = max(watched_genre_list, key = watched_genre_list.count)
```

Reply:
This is something to talk about our next 1 on 1 meeting I think, because I don't understand how this max function is working and google is failing me.

Reply:
I'm going to edit my comment above, I was grading a little too late and messed up the key portion above (get is applicable to dictionaries, we want to use count for a list).

Reply:
Here's a resource on `max` with some examples: https://www.programiz.com/python-programming/methods/built-in/max and we can definitely talk about it when we meet!

Reply:
Okay that helps considerably!!! I was searching for just max and key and it just was not clicking. Also thank you for explaining that the function considers the letters that come last in the alphabet as "high" because I kept seeing examples of people using max on strings without clarifying that point and could not figure out why the results were what they were. 
So with key, you can specify something done to the iterable _before_ the max function runs, and it compares those results but still returns the actual list value as opposed to what the key did, correct? So in this case, we tell it to count each item on the list and find the highest count, but still return whatever the original value we were counting was.

Reply:
Nice, I'm glad that helped. And your summary is nearly spot on; what we pass in for key isn't done to the iterable _before_ max, max will call whatever we've passed for the key during it's execution. Everything after that is correct, max will compare the results of running our key function (count) on each item in the list, then return the list item that corresponded with the highest value our key function (count) gave.